### PR TITLE
Mhv 59878 pilot api feature flag

### PR DESCRIPTION
**Pull request by fazilqa**
## Summary

- This PR adds new scenario for `pilot env` inbox view where user can see both MHV and OH messages. 

## Related issue(s)

- https://jira.devops.va.gov/browse/MHV-59878
- https://jira.devops.va.gov/browse/MHV-53916

## Testing done

- Relevant tests have been executed in headed mode with a 200x pattern.
- All tests have been successfully conducted in headless mode without any failures.

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv-secure-messaging/tests/e2e/pages/PilotEnvPage.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import mockPilotMessages from '../fixtures/pilot-responses/inbox-threads-OH-response.json';
+import mockFolders from '../fixtures/pilot-responses/folders-respose.json';
+import { Paths } from '../utils/constants';
+
+class PilotEnvPage {
+  loadInboxMessages = (
+    num = 1,
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/e2e/pages/PilotEnvPage.js:7]: not sure if this argument is needed as this always will be 1 in PilotEnvPage class, isn't it?

File: src/applications/mhv-secure-messaging/tests/e2e/pilot-env-tests/secure-messaging-pilot-inbox.cypress.spec.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    PilotEnvPage.loadInboxMessages();
+
+    PilotEnvPage.verifyUrl(Paths.UI_PILOT);
+    PilotEnvPage.verifyThreadLength(mockMessages);
+
+    cy.injectAxe();
+    cy.axeCheck(AXE_CONTEXT);
+  });
+
+  it('pilot OF inbox page view', () => {
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/e2e/pilot-env-tests/secure-messaging-pilot-inbox.cypress.spec.js:38]: typo here
```suggestion
  it('pilot OFF inbox page view', () => {
```

Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/e2e/pilot-env-tests/secure-messaging-pilot-inbox.cypress.spec.js:38]: is this supposed to be OFF or ON

File: src/applications/mhv-secure-messaging/tests/e2e/fixtures/pilot-responses/inbox-threads-OH-response.json [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        "messageId": 3764003,
+        "threadPageSize": 3,
+        "messageCount": 5,
+        "category": "OTHER",
+        "subject": "OH THREAD 002 ATTACHMENT",
+        "triageGroupName": "OH TG GROUP 002 ",
+        "sentDate": "2024-07-10T18:53:28.000Z",
+        "draftDate": null,
+        "senderId": 2444644,
+        "senderName": "HAMLAT, ABDELHALIM",
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/e2e/fixtures/pilot-responses/inbox-threads-OH-response.json:18]: May need to change this PII with fake data. there is more below

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]
@fazilqa I left some review comments that may be worth addressing

File: src/applications/mhv-secure-messaging/tests/e2e/fixtures/pilot-responses/inbox-threads-OH-response.json (hunk 3 above)
Comment (fazilqa) [reply src/applications/mhv-secure-messaging/tests/e2e/fixtures/pilot-responses/inbox-threads-OH-response.json:18]: done

**Comment by fazilqa** [comment]
> @fazilqa I left some review comments that may be worth addressing

all fixed, thanks for mentioning.

File: src/applications/mhv-secure-messaging/tests/e2e/fixtures/pilot-responses/inbox-threads-response.json [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      "type": "message_threads",
+      "attributes": {
+        "threadId": 3763944,
+        "folderId": 0,
+        "messageId": 3763954,
+        "threadPageSize": 3,
+        "messageCount": 2,
+        "category": "TEST_RESULTS",
+        "subject": "MHV THREAD 002  ATTACH",
+        "triageGroupName": "Abdel Triage DAYT29",
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/e2e/fixtures/pilot-responses/inbox-threads-response.json:14]: ☝🏻

File: src/applications/mhv-secure-messaging/tests/e2e/fixtures/pilot-responses/inbox-threads-response.json [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        "sentDate": "2024-07-10T18:49:12.000Z",
+        "draftDate": null,
+        "senderId": 2444644,
+        "senderName": "HAMLAT, ABDELHALIM",
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/e2e/fixtures/pilot-responses/inbox-threads-response.json:18]: ☝🏻

File: src/applications/mhv-secure-messaging/tests/e2e/fixtures/pilot-responses/inbox-threads-response.json [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+      "type": "message_threads",
+      "attributes": {
+        "threadId": 3763921,
+        "folderId": 0,
+        "messageId": 3763930,
+        "threadPageSize": 3,
+        "messageCount": 3,
+        "category": "TEST_RESULTS",
+        "subject": "MHV THREAD 001",
+        "triageGroupName": "Abdel Triage DAYT29",
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/e2e/fixtures/pilot-responses/inbox-threads-response.json:41]: ☝🏻

File: src/applications/mhv-secure-messaging/tests/e2e/fixtures/pilot-responses/inbox-threads-response.json [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        "sentDate": "2024-07-10T18:47:02.000Z",
+        "draftDate": null,
+        "senderId": 2444644,
+        "senderName": "HAMLAT, ABDELHALIM",
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/e2e/fixtures/pilot-responses/inbox-threads-response.json:45]: ☝🏻

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]
couple more places we need to clean up PII. Also, if I am not mistaken, wea re not adding coverage for an individual message thread and search, would you please confirm?

**Comment by fazilqa** [comment]
> couple more places we need to clean up PII. Also, if I am not mistaken, wea re not adding coverage for an individual message thread and search, would you please confirm?

confirmed.  I can cover it in further tickets

**Review by alex-morgun-1 — APPROVED** [review]

**Comment by oleksii-morgun** [comment]
> > couple more places we need to clean up PII. Also, if I am not mistaken, wea re not adding coverage for an individual message thread and search, would you please confirm?
> 
> confirmed. I can cover it in further tickets

Let's ensure we have tickets created in that case

**Pull request merged**